PR_REVIEW_CHECKLIST.md

```diff
@@ -0,0 +1,98 @@
+# PR review checklist
+
+This document lists specific checks that should be done when reviewing a PR. This means these items should be checked by:
+1. Every contributor submitting a Pull Request
+2. Every C+ (Contributor Plus) team member reviewing a PR
+3. Every Expensify employee submitting / reviewing a PR
```

Comment:
I see what you mean, but I think it's nice to be explicit here so that it's **clear** to contributors & C+ that they need to know & be following this checklist all the time.